lib/recurly/transaction.rb

```diff
@@ -125,5 +125,36 @@ def inspect
       end
       super attributes
     end
+
+    # https://github.com/recurly/recurly-client-ruby/pull/436
+    def marshal_dump
+      [
+        @attributes.reject { |k, v| v.is_a?(Proc) },
+        @new_record,
+        @destroyed,
+        @uri,
+        @href,
+        changed_attributes,
+        previous_changes,
+        response,
+        etag,
+        links,
+        @type
+      ]
+    end
+
+    def marshal_load(serialization)
```

Comment:
I think it should be safe to add `@type` to the list in `Resource`'s definition. It will just be nil for any object serialized with a previous version.

Reply:
So we probably don't need this method here.

Reply:
Do you think we will get any benefit from that except for less code?

Reply:
Ok, I get it. The only fear is that this extra type instance variable after marshaling could cause some other weird issue.